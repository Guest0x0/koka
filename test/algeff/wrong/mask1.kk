// issue #509

effect reader<s>
  fun ask() : s

fun test( p : () -> <reader<int>, reader<int>|e> a ) : e a
  with fun ask() 100
  with fun ask() 42
  p()

pub fun main(): console ()
  with test
  println(mask<reader<int>>(ask()) : int)