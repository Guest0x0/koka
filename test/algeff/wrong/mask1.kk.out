test/algeff/wrong/mask1@kk(13,29): type error: types do not match (due to an infinite type)
 context : mask<reader<int>>(ask()
 term : ask()
 inferred effect: <reader<() -> _e int>|_e1>
 expected effect: _e
 because : inject expects a parameter-less function argument
 hint : give a type to the function definition?